### PR TITLE
nuScenes CAN bus extension

### PR DESCRIPTION
**Pull request by holger-nutonomy**
This PR augments the nuScenes dataset with the CAN vehicle bus data extracted for each scene. The files are available on https://www.nuscenes.org/download shortly. This PR contains:
- A readme.
- A Jupyter notebook tutorial.
- A simple API to read the JSON files that contain the CAN bus messages.

Below are a few renderings from the tutorial:
Vehicle pose vs. route:
download

Steering angle profile over time:
download (2)

Lateral acceleration profile over time:
download (1)

### CODE REVIEW

File: python-sdk/nuscenes/can_bus/README.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+- q: \[4\] Quaternion that transforms from IMU coordinates to a fixed reference frame. The yaw of this reference frame is arbitrary, determined by the IMU. However, the x-y plane of the reference frame is perpendicular to gravity, and z points up. 
+- rotation_rate: \[3\] Angular velocity in rad/s around the x, y, and z axes, respectively, in the IMU coordinate frame.
+
+### Pose
+(50Hz, *scene_%04d_pose.json*)
+
+The current pose of the ego vehicle, sampled at 50Hz.
+- accel: \[3\] Acceleration vector in the ego vehicle frame in m/s/s.
+- orientation: \[4\]  The rotation vector in the ego vehicle frame.
+- pos: \[3\] The position (x, y, z) in meters in the ego vehicle frame. This should be identical to the [nuScenes ego pose](https://github.com/nutonomy/nuscenes-devkit/blob/master/schema.md#ego_pose), but sampled at a higher frequency.
```
Comment (freddyaboulton) [inline python-sdk/nuscenes/can_bus/README.md]: Isn't this the global frame?

File: python-sdk/nuscenes/can_bus/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+- battery_level: \[1\] Current battery level in range \[0, 100\].
+- brake: \[1\] Braking pressure in bar. An integer in range \[0, 126\]. 
+- brake_switch: \[1\] Brake switch as an integer, 1 (pedal not pressed), 2 (pedal pressed) or 3 (pedal confirmed pressed).
+- gear_position: \[1\] The gear position as an integer, typically 0 (parked) or 7 (driving).
+- left_signal: \[1\] Left turning signal as an integer, 0 (inactive) or 1 (active).
+- rear_left_rpm: \[1\] Rear left brake speed in revolutions per minute.
+- rear_right_rpm: \[1\] Rear right brake speed in revolutions per minute.
+- right_signal: \[1\] Right turning signal as an integer, 0 (inactive) or 1 (active).
+- steering: \[1\] Steering angle in degrees at a resolution of 0.1 in range \[-780, 779.9\].
+- steering_speed: \[1\] Steering speed in degrees per second.
```
Comment (freddyaboulton) [inline python-sdk/nuscenes/can_bus/README.md]: Is this radians / second ?

File: python-sdk/nuscenes/can_bus/README.md (hunk 1 above)
Comment (holger-nutonomy) [reply python-sdk/nuscenes/can_bus/README.md]: Good find!

File: python-sdk/nuscenes/can_bus/README.md (hunk 2 above)
Comment (holger-nutonomy) [reply python-sdk/nuscenes/can_bus/README.md]: Nope, degrees / second. I've added the min and max to show that it is signed.

**Pull request merged**